lib/oneview-sdk/resource/api500/c7000/volume.rb

```diff
@@ -151,6 +174,15 @@ def get_volume_template_uri(template_data)
         def generate_snapshot_data(name, description = nil)
           { description: description, name: name }
         end
+
+        # Gets the volume
+        # @raise [OneviewSDK::IncompleteResource] if the name parameter is not set
+        # @return [Array] the array of volumes
+        def find
+          @data['properties'] = Hash[@data['properties'].map { |k, v| [k.to_s, v] }]
```

Comment:
Do we really want to be converting the whole `@data['properties']` here?
This is likely less expensive and will accomplish the same thing:
```ruby
name = @data['name'] || @data[:name]
```